pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -73,959 +71,762 @@
-            if self.num_processes is None:
-                # define the max CPU available
-                self.num_processes = os.cpu_count()
-        # special case with TPUs
-        elif self.has_tpu and not _use_cpu:
-            self._device_type = _AcceleratorType.TPU
-            if isinstance(self.tpu_cores, int):
-                self._strategy_type = _StrategyType.TPU_SPAWN
-        elif self.has_ipu and not _use_cpu:
-            self._device_type = _AcceleratorType.IPU
-        elif self.distributed_backend and self._strategy_type is None:
-            self._strategy_type = _StrategyType(self.distributed_backend)
-
-        if self.num_gpus > 0 and not _use_cpu:
-            self._device_type = _AcceleratorType.GPU
-
-        _gpu_strategy_types = (_StrategyType.DP, _StrategyType.DDP, _StrategyType.DDP_SPAWN, _StrategyType.DDP2)
-        # DP and DDP2 cannot run without GPU
-        if self.num_gpus == 0 and self._strategy_type in _gpu_strategy_types and not _use_cpu:
-
-            if (self.num_nodes and self.num_nodes > 1) or (self.num_processes and self.num_processes > 1):
-                if self._strategy_type in (_StrategyType.DP, _StrategyType.DDP2):
-                    rank_zero_warn(
-                        f"{self._strategy_type.value!r} is not supported on CPUs, hence setting `strategy='ddp'`."
-                    )
-                    self._strategy_type = _StrategyType.DDP
-            else:
-                rank_zero_warn("You are running on single node with no parallelization, so distributed has no effect.")
-                self._strategy_type = None
+    # def _distrib_type():
+    @property
+    def device_type(self):
+        if isinstance(self.accelerator, CPUAccelerator):
+            return "cpu"
+        if isinstance(self.accelerator, GPUAccelerator):
+            return "gpu"
+        if isinstance(self.accelerator, TPUAccelerator):
+            return "tpu"
+        if isinstance(self.accelerator, IPUAccelerator):
+            return "ipu"
 
-        # finished configuring self._strategy_type, check ipython environment
-        self.check_interactive_compatibility()
+    @property
+    def num_nodes(self):
+        return self._num_nodes_flag
 
-        # for DDP overwrite nb processes by requested GPUs
-        if self._device_type == _AcceleratorType.GPU and self._strategy_type in (
-            _StrategyType.DDP,
-            _StrategyType.DDP_SPAWN,
-        ):
-            self.num_processes = self.num_gpus
+    @property
+    def num_processes(self):
+        return self.devices if self.devices is not None else 1
 
-        if self._device_type == _AcceleratorType.GPU and self._strategy_type == _StrategyType.DDP2:
-            self.num_processes = self.num_nodes
+    @property
+    def root_gpu(self) -> Optional[int]:
+        return (
+            self.strategy.root_device.index
+            if not isinstance(self.accelerator, (IPUAccelerator, TPUAccelerator))
+            else None
+        )
```

Comment:
why do we need to expose root_gpu here when there's strategy.root_device 😠

Reply:
All properties after line 751 is for backward compatibility :)
```
 ##############################################################################
    # the following logic should be deprecated/removed, and these information should be
    # retrive from strategies and accelerators
    # Added here to keep backward compabilities
```